salt/master.py

```diff
@@ -258,7 +258,16 @@ def run(self):
         now = int(time.time())
 
         git_pillar_update_interval = self.opts.get("git_pillar_update_interval", 0)
-        old_present = set()
+        # load the presence data from the cache on disk
+        presence_cache = salt.utils.cache.CacheFactory.factory(
+            "disk",
+            3600,
```

Comment:
the value of 3600 doesn't really matter all that much.. theoretically we could increase it to handle longer shutdowns of the salt master but that's probably an edge case, and from my testing I think there is  bug in the current `CacheDisk` class anyway where I think the `CacheDisk_cachetime` key is being stored as the human readable but completely valueless `present` rather than an actual timestamp. I'd rather not have to fix that myself here.

Reply:
Ah ok there is no bug. I just was seeign the key rather than the value. All good here.